### DISCUSSION TITLE
Database connection pool size in legacy Play 2.3 app

### DISCUSSION THREAD

**Discussion by nilsweinander** [playframework/playframework#12330] [Q&A] [answered]
I have recently had problems with database connections in a Play 2.3 app, getting

`
Timed out waiting for a free available connection.
`

errors from getConnection(). So I thought to increase the connection pool size and found this in the documentation:

```
# In order to reduce lock contention and thus improve performance, 
# each incoming connection request picks off a connection from a 
# pool that has thread-affinity. 
# The higher this number, the better your performance will be for the 
# case when you have plenty of short-lived threads. 
# Beyond a certain threshold, maintenance of these pools will start 
# to have a negative effect on performance (and only for the case 
# when connections on a partition start running out).
db.default.partitionCount=2

# The number of connections to create per partition. Setting this to 
# 5 with 3 partitions means you will have 15 unique connections to the 
# database. Note that BoneCP will not create all these connections in 
# one go but rather start off with minConnectionsPerPartition and 
# gradually increase connections as required.
db.default.maxConnectionsPerPartition=5

# The number of initial connections, per partition.
db.default.minConnectionsPerPartition=5

```

But, I cannot find any information about the default values, so it is hard to know if I actually do increase pool size if I add these parameters to configuration. Is there any way to find the default settings?

**Comment by mkurz** [comment, marked as answer]
> Is there any way to find the default settings?

For Play 2.3:
https://github.com/playframework/playframework/blob/2.3.x/framework/src/play-jdbc/src/main/scala/play/api/db/DB.scala#L378
Starting with Play 2.4 you find all the default configs in various `reference.conf` files, e.g. in your case:
https://github.com/playframework/playframework/blob/2.4.x/framework/src/play-jdbc/src/main/resources/reference.conf

**Reply by nilsweinander** [reply]
Thanks! Exactly what I needed!
